dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +53,176 @@ public Pattern getParameterPattern() {
         return paramRegex;
     }
 
-    public Map<String, DbParameterAccessor> getAllColumns(
-            final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
-                + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
-                + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+    public Map<String, DbParameterAccessor> getAllColumns(String tableName)
+                                                        throws SQLException {
+        DatabaseTable t = new DatabaseTable(getConnection(), getConnection().getSchema(), tableName.toUpperCase());
+        return t.getParams();
     }
 
-    private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
-        try (PreparedStatement dc = getConnection().prepareStatement(query)) {
-            dc.setString(1, tableOrViewName);
+    private abstract class DatabaseObject {
+        protected Connection con;
```

Comment:
I can get at that method with `DerbyEnvironment.this.getConnection()` but then end up with long expressions.
E.g.
`DerbyEnvironment.this.getConnection().getMetaData().getFunctionColumns(null, name.getSchemaName(), name.getObjectName(), "%");`

What preferred practice here? Introduce another variable to reference the result of ``DerbyEnvironment.this.getConnection()` or just break up the expression over multiple lines. E.g.

```
ResultSet rs = DerbyEnvironment.this.getConnection().getMetaData()
                   .getFunctionColumns(null, name.getSchemaName(),
                   name.getObjectName(), "%");
```